File: frontend/src/types/quiz.ts
export type QuizType =
  | "Multiple Choice"
  | "True or False"
  | "Identification"
  | "Open Ended"
  | "Mixed Assessment";

export type UserRole = "teacher" | "student";
export type SourceMaterialType = "topic" | "image" | "pdf" | "text";

export interface CvSUUser {
  id: number;
  email: string;
  full_name: string;
  picture_url?: string;
  role: UserRole;
  token?: string;
}

export interface GoogleAuthPayload {
  credential?: string;
  email?: string;
  name?: string;
  picture?: string;
  role: UserRole;
  is_simulation?: boolean;
}

export interface QuizSection {
  quiz_type: QuizType;
  question_count: number;
}

export interface QuizPublishPayload {
  topic?: string;
  question_count: number;
  quiz_type: QuizType;
  sections?: QuizSection[];
  created_by_email?: string;
  source_type?: "topic" | "file" | "text";
  content_text?: string;
  file_base64?: string;
  file_mime_type?: string;
  file_name?: string;
}

export interface Question {
  question: string;
  options: string[];
  answer: string;
  quiz_type?: QuizType;
  section_name?: string;
}

export interface StudentQuestion {
  question: string;
  options: string[];
  quiz_type?: QuizType;
  section_name?: string;
}

export interface PublishedQuiz {
  id: string;
  topic: string;
  quiz_type: QuizType;
  question_count: number;
  created_by_email?: string;
  created_at: string;
}

export interface QuizDetail {
  id: string;
  topic: string;
  quiz_type: QuizType;
  question_count: number;
  created_by_email?: string;
  created_at?: string;
  questions: StudentQuestion[];
}

export interface GradeSubmission {
  id?: number;
  student_name: string;
  student_email?: string;
  quiz_id: string;
  topic: string;
  score: number;
  total_questions: number;
  percentage: number;
  answers: AnswerDetail[];
  created_at?: string;
}

export interface AnswerDetail {
  question: string;
  user_answer: string;
  correct_answer: string;
  is_correct: boolean;
  ai_feedback?: string;
}

export interface GradeResult {
  is_correct: boolean;
  score?: number;
  feedback: string;
}

File: backend/app/schemas/quiz.py
from datetime import datetime
from pydantic import BaseModel
from typing import Any

class QuizSection(BaseModel):
    quiz_type: str
    question_count: int

class QuizRequest(BaseModel):
    topic: str | None = None
    question_count: int | None = None
    quiz_type: str | None = None
    sections: list[QuizSection] | None = None
    created_by_email: str | None = None
    source_type: str | None = None
    content_text: str | None = None
    file_base64: str | None = None
    file_mime_type: str | None = None
    file_name: str | None = None

class QuizSubmissionRequest(BaseModel):
    student_name: str
    student_email: str
    quiz_id: str
    topic: str
    score: float
    total_questions: int
    answers: list[dict[str, Any]]

class GradeAnswerRequest(BaseModel):
    question: str
    correct_answer: str
    user_answer: str
    quiz_type: str

class QuizSummaryResponse(BaseModel):
    id: str
    topic: str
    quiz_type: str
    question_count: int
    created_by_email: str | None
    created_at: datetime

class StudentQuestionResponse(BaseModel):
    question: str
    options: list[str] | list[dict[str, Any]] | None = None
    quiz_type: str
    section_name: str | None = None

class QuizDetailResponse(BaseModel):
    id: str
    topic: str
    quiz_type: str
    question_count: int
    created_by_email: str | None
    created_at: datetime
    questions: list[StudentQuestionResponse]
